File: Numeros Pseudo-Aleatorios/Linear Congruential/LinearCongruential.scala
/*
 PRNG .......: Linear Congruential
 Ano ........: 1965
 Autor ......: W. E. Thomson
*/

import java.util.Date

object Main {
  def linearCongruential(seed: Array[Long]): Double = {
    val M: Long = 2147483647
    seed(0) = (1664525 * seed(0) + 1013904223) % M
    seed(0).toDouble / M
  }

  def main(args: Array[String]): Unit = {
    val Xn1: Array[Long] = Array(1234)
    val Xn2: Array[Long] = Array(new Date().getTime)

    println("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)")
    println("PSEUDORANDOM NUMBER GENERATOR (PRNG)")
    println("PADRAO: Linear Congruential")
    println("==========================================")
    println()
    println("    Semente: 1234        Semente: TIME")
    println("    -------------        -------------")

    for (i <- 1 to 10) {
      printf("%17.11f        %13.11f%n", linearCongruential(Xn1), linearCongruential(Xn2))
    }
  }
}

File: Numeros Pseudo-Aleatorios/Linear Congruential/LinearCongruential.java
import java.util.Date;

public class Main {

    static double linearCongruential(long[] seed) {
        final long M = 2147483647;
        seed[0] = (1664525 * seed[0] + 1013904223) % M;
        return (double) seed[0] / M;
    }

    public static void main(String[] args) {
        long[] Xn1 = { 1234 };
        long[] Xn2 = { new Date().getTime() };

        System.out.println("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)");
        System.out.println("PSEUDORANDOM NUMBER GENERATOR (PRNG)");
        System.out.println("PADRAO: Linear Congruential");
        System.out.println("==========================================");
        System.out.println();
        System.out.println("    Semente: 1234        Semente: TIME");
        System.out.println("    -------------        -------------");

        for (int i = 1; i <= 10; i++) {
            System.out.printf("%17.11f        %13.11f%n",
                    linearCongruential(Xn1), linearCongruential(Xn2));
        }

    }

}
